### DISCUSSION TITLE
Preferred configuration for resistive touch with Universal Touch

### DISCUSSION THREAD

**Discussion by s-hadinger** [arendst/Tasmota#21020] [Ideas]
@gemu2015 I started using the Universal Touch driver, awesome work!

I got back my Lolin TFT that uses the `XPT2046` driver. It worked fine except that the `x` was reversed. The `SCL` opcode does not make it very friendly to reverse the scale, it needs some extra calculation.

I already did the `ts_calibrate.be` app to help easy calibration of resistive touchscreen that generates the `:M` line. I see that this still works with universal touch.

My question: don't you think it would be easier for calibration to use a combination of `:UTI` and `:M`, instead of just using `SCL` directive?

Im my case is would be:
```
:M,3931,378,185,3813           <- use previous :M section
:UTI,XPT2046,S1,27,-1,-1
CPR 00
RT
:UTT
XPT 300
RT
:UTX
MV 0 2            <- no SCL
RT
:UTY
MV 2 2            <- no SCL
RT
#
```

instead of the example in the doc:
```
:UTI,XPT2046,S1,21,-1,22
CPR 00
RT
:UTT
XPT 300
RT
:UTX
MV 0 2
;DBG 1
SCL 460 0.0930233
LIM 319
RT
:UTY
MV 2 2
SCL 300 0.0695652
LIM 239
RT
#
```

**Comment by gemu2015** [comment]
yes you can use the M: as usual,
but i have a display that can not be made working with the M: parameters.
therefore i created the SCL opcode which allows any combination.

we should use M: when possible and only use SCL when M: does not fit.
